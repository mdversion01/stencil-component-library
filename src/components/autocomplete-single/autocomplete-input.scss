// AutocompleteInput styles
*,
::after,
::before {
  box-sizing: border-box;
}

.autocomplete-container {
  .input-group {
    flex-wrap: nowrap !important;
  }

  .autocomplete-single-select {
    border: 1px solid #ced4da;
    border-radius: 0.25rem;

    .form-control,
    .form-control:focus {
      background-color: transparent;
      border: none !important;
      color: #383838;
      outline: 0 !important;
      box-shadow: none !important;
    }

    >.input-group-btn {
      border: none;
    }

    &.ac-focused {
      border-color: #9bc4f6 !important;
      box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25) !important;
    }

    &.is-invalid-focused {
      border-color: #dc3545 !important; /* Fallback for --bs-form-invalid-border-color */
      box-shadow: 0 0 0 0.25rem rgba(220, 53, 69, 0.25) !important; /* Fallback for --bs-danger-rgb */
    }

    &.is-invalid {
      border-color: #b30009;
    }

    .input-group-btn.clear svg:hover {
      color: #b80000;
      fill: #b80000;
      outline: none;
    }

    .input-group-btn.clear svg:focus-visible {
      color: #b80000;
      fill: #b80000;
      outline: 1px solid #b80000;
    }

    &.is-invalid>.input-group-btn.add,
    &.is-invalid > .input-group-btn.clear {
      color: #b30009;
    }

    &.is-invalid >.input-group-btn.add svg,
    &.is-invalid > .input-group-btn.clear svg {
      color: #b30009;
      fill: #b30009;
    }

    .input-group-btn.add,
    .input-group-btn.add svg {
      color: #006108;
      fill: #006108;
    }

    .input-group-btn.add i:hover,
    .input-group-btn.add svg:hover {
      color: #008f0c;
      fill: #008f0c;
      outline: none;
    }

    .input-group-btn.add i:focus-visible,
    .input-group-btn.add svg:focus-visible {
      color: #008f0c;
      fill: #008f0c;
      outline: 1px solid #008f0c;
    }

    &.is-invalid > .input-group-btn.add {
      color: #b30009;
    }
  }
}

.autocomplete-dropdown {
  position: absolute;
  min-width: 150px;
  max-width: 300px;
  max-height: 200px;
  overflow-y: auto;
  background-color: #fff;
  border: 1px solid #ccc;
  margin-top: 1px;
  z-index: 1000;
}

.autocomplete-dropdown ul {
  list-style: none;
  margin: 0;
  padding: 0;
}

.autocomplete-dropdown-item {
  padding: 5px 10px;
  cursor: pointer;
}
.autocomplete-dropdown-item:hover {
  background-color: #f2f2f2;
}

.autocomplete-dropdown-item:active {
  background-color: #2680eb;
  color: #fff;
  text-decoration: none;
}

.autocomplete-dropdown-item.focused {
  background-color: #eee;
}

.autocomplete-dropdown-item.key-selected {
  background-color: #5a5a5a;
  color: #fff;
}

.autocomplete-dropdown-item.key-selected.focused {
  background-color: #3a3a3a;
  color: #fff;
}

.autocomplete-dropdown-no-results {
  padding: 0.5rem 1rem;
  color: #888;
  font-style: italic;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;

  &.sm {
    font-size: 0.75rem;
  }

  &.lg {
    font-size: 1.2rem;
  }
}

.ac-combobox-container {
  margin-bottom: 0.25rem;
  position: relative;
}

.ac-multi-select-container .pl-input-group-append .pl-input-group-btn {
  border: none;
}

.ac-basic-container,
.ac-input-container {
  display: flex;
  flex: 1 1 auto;
}

.autocomplete-dropdown-item.sm {
  font-size: 0.75rem;
}

.autocomplete-dropdown-item.lg {
  font-size: 1.2rem;
}

.button-brdr {
  border-right: none;
}

.error-message {
  color: red;
  margin-top: 4px;
  font-size: 11px;
}

@media (max-width: 365px) {
  .was-validated .form-control:invalid,
  .form-control.is-invalid {
    background-image: none !important;
  }
}

.input-group-append {
  height: 100%;
  position: relative;
  z-index: 5;
}

.form-group {
  position: relative;
}
