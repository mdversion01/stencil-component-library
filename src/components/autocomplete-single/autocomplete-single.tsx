// src/components/autocomplete-single/autocomplete-single.tsx
import { Component, h, Prop, State, Element, EventEmitter, Event, Watch } from '@stencil/core';
import { logInfo, logWarn, logError } from '../../utils/log-debug';

@Component({
  tag: 'autocomplete-single',
  styleUrls: [
    '../layout-styles.scss',
    '../form-styles.scss',
    '../input-field/input-field-styles.scss',
    '../input-group/input-group-styles.scss',
    './autocomplete-input.scss',
  ],
  shadow: false,
})
export class AutocompleteSingle {
  @Element() el: HTMLElement;

  @Prop({ mutable: true }) options: string[] = [];
  @Prop() addBtn = false;
  @Prop() addIcon = '';
  @Prop() arialabelledBy: string = '';
  @Prop() clearIcon = '';
  @Prop() placeholder: string = 'Type to search/filter...';
  @Prop() devMode: boolean = false;
  @Prop() disabled: boolean = false;
  @Prop() formId: string = '';
  @Prop() formLayout: string = ''; // 'horizontal'
  @Prop({ mutable: true }) error = false;
  @Prop({ mutable: true }) errorMessage = '';
  @Prop() inputId = '';
  @Prop() label: string = '';
  @Prop() labelHidden: boolean = false;
  @Prop() removeClearBtn: boolean = false;
  @Prop() size: string = ''; // 'sm' | 'lg'
  @Prop() required: boolean = false;
  @Prop() type = '';
  @Prop({ mutable: true }) validation: boolean = false;
  @Prop() validationMessage: string = '';

  @State() inputValue: string = '';
  @State() filteredOptions: string[] = [];
  @State() focusedOptionIndex: number = -1;
  @State() selectedOptionIndex: number = -1;
  @State() isFocused: boolean = false;
  @State() hasBeenInteractedWith: boolean = false;
  @State() dropdownOpen: boolean = false;

  @Event({ eventName: 'itemSelect' }) itemSelect: EventEmitter<string>;
  @Event() clear: EventEmitter<void>;
  @Event() componentError: EventEmitter<{ message: string; stack?: string }>;

  @Watch('options')
  handleOptionsChange(newVal: string[]) {
    if (!Array.isArray(newVal)) {
      logError(this.devMode, 'AutocompleteSingle', `'options' should be an array`, { receivedType: typeof newVal });
      return;
    }
    this.filterOptions();
  }

  private handleFocus = () => {
    this.isFocused = true;
  };

  private handleBlur = () => {
    this.isFocused = false;
  };

  private camelCase(str: string): string {
    return str.replace(/(?:^\w|[A-Z]|\b\w)/g, (word, index) => (index === 0 ? word.toLowerCase() : word.toUpperCase())).replace(/\s+/g, '');
  }

  private sanitizeInput(value: string): string {
    return value.replace(/[<>]/g, '');
  }

  private handleInput = (event: InputEvent | KeyboardEvent) => {
    const input = event.target as HTMLInputElement;

    if (event.type === 'keydown') {
      const key = (event as KeyboardEvent).key;
      if (key === 'ArrowDown') {
        event.preventDefault();
        this.navigateOptions(1);
      } else if (key === 'ArrowUp') {
        event.preventDefault();
        this.navigateOptions(-1);
      } else if (key === 'Enter' && this.focusedOptionIndex >= 0) {
        this.selectOption(this.filteredOptions[this.focusedOptionIndex]);
      } else if (key === 'Escape') {
        this.closeDropdown();
      }
    } else {
      this.inputValue = this.sanitizeInput(input.value);
      this.filterOptions();
      this.hasBeenInteractedWith = true; // ✅ mark as touched

      // if (this.required && this.inputValue.trim() === '') {
      //   this.validation = true; // ✅ restore validation only if required
      // }
    }
  };

  private filterOptions() {
    if (!Array.isArray(this.options)) {
      logError(this.devMode, 'AutocompleteSingle', `'options' must be an array`, {
        receivedType: typeof this.options,
        value: this.options,
      });
      this.filteredOptions = [];
      this.dropdownOpen = false;
      return;
    }

    if (this.inputValue.length > 0) {
      this.filteredOptions = this.options.filter(opt =>
        opt.toLowerCase().includes(this.inputValue.toLowerCase())
      );
      this.dropdownOpen = true;
    } else {
      this.filteredOptions = [];
      this.dropdownOpen = false;
    }
  }

  private ensureOptionInView(index: number) {
    setTimeout(() => {
      const items = this.el.querySelectorAll('.autocomplete-dropdown-item');
      if (items && index >= 0 && index < items.length) {
        const item = items[index] as HTMLElement;
        item.scrollIntoView({ block: 'nearest', behavior: 'smooth' });
      }
    }, 0);
  }

  private navigateOptions(direction: number) {
    const newIndex = this.focusedOptionIndex + direction;

    if (!Array.isArray(this.filteredOptions)) {
      logError(this.devMode, 'AutocompleteSingle', `'filteredOptions' is not an array`, {
        receivedType: typeof this.filteredOptions,
      });
      return;
    }

    if (newIndex >= 0 && newIndex < this.filteredOptions.length) {
      this.focusedOptionIndex = newIndex;
      this.ensureOptionInView(newIndex);
    } else {
      logWarn(this.devMode, 'AutocompleteSingle', `Navigation index out of bounds`, {
        attemptedIndex: newIndex,
        totalOptions: this.filteredOptions.length,
      });
    }
  }

  private renderLabel(ids: string) {
    if (!this.label) return null;
    return (
      <label
        htmlFor={ids}
        class={{
          'form-label': true,
          'sr-only': this.labelHidden,
          'invalid': this.validation,
          'required-text': this.validation,
          'col-sm-2 col-form-label': this.formLayout === 'horizontal' || this.formLayout === 'inline',
        }}
      >
        <span>
          {this.label}
          {this.required ? <span class="required">*</span> : ''}
        </span>
      </label>
    );
  }

  private renderInputField(ids: string, names: string) {
    return (
      <input
        id={ids ? ids : null}
        name={names ? names : null}
        role="combobox"
        aria-label={this.labelHidden ? names : null}
        aria-labelledby={this.arialabelledBy}
        aria-describedby={this.validation ? `${ids}-validation` : this.error ? `${ids}-error` : null}
        aria-autocomplete="list"
        aria-expanded={this.filteredOptions.length > 0 ? 'true' : 'false'}
        aria-controls={`${ids}-listbox`}
        aria-activedescendant={this.focusedOptionIndex >= 0 ? `${ids}-option-${this.focusedOptionIndex}` : undefined}
        aria-required={this.required ? 'true' : 'false'}
        aria-haspopup="listbox"
        class={{
          'form-control': true,
          'is-invalid': this.validation,
          [`input-${this.size}`]: !!this.size,
        }}
        type={this.type}
        placeholder={this.placeholder}
        value={this.inputValue}
        disabled={this.disabled}
        onInput={this.handleInput}
        onKeyDown={this.handleInput}
        onFocus={this.handleFocus}
        onBlur={this.handleBlur}
        inputmode="text"
        autocomplete="off"
        spellcheck="false"
      />
    );
  }

  private renderDropdownList(ids: string) {
    return (
      <ul role="listbox" id={`${ids}-listbox`} tabindex="-1">
        {this.filteredOptions.length > 0 ? (
          this.filteredOptions.map((option, index) => (
            <li
              id={`${ids}-option-${index}`}
              role="option"
              aria-selected={this.focusedOptionIndex === index ? 'true' : 'false'}
              class={{
                'autocomplete-dropdown-item': true,
                'focused': this.focusedOptionIndex === index,
                [`${this.size}`]: !!this.size,
              }}
              onClick={() => this.selectOption(option)}
              tabindex="-1"
            >
              <span innerHTML={option.replace(/</g, '&lt;').replace(/>/g, '&gt;')} />
            </li>
          ))
        ) : (
          <li
            class={{
              'autocomplete-dropdown-no-results': true,
              [`${this.size}`]: !!this.size,
            }}
            aria-disabled="true"
          >
            No results found
          </li>
        )}
      </ul>
    );
  }

  private renderDropdown(ids: string) {
    if (!this.dropdownOpen) return null;

    return (
      <div class="autocomplete-dropdown" aria-live="polite">
        {this.renderDropdownList(ids)}
      </div>
    );
  }

  private selectOption(option: string) {
    if (typeof option !== 'string' || option.trim() === '') {
      logError(this.devMode, 'AutocompleteSingle', 'Invalid option selected', { option });
      return;
    }

    this.inputValue = option;
    this.filteredOptions = [];
    this.focusedOptionIndex = -1;
    this.selectedOptionIndex = this.options.indexOf(option);
    this.validation = false;
    this.itemSelect.emit(option);
    this.dropdownOpen = false;

    // ✅ Restore focus to input after selecting
    setTimeout(() => {
      const input = this.el.querySelector('input');
      input?.focus();
    }, 0);

    logInfo(this.devMode, 'AutocompleteSingle', 'Item selected', { selected: option });
  }

  private closeDropdown() {
    this.filteredOptions = [];
    this.focusedOptionIndex = -1;

    logInfo(this.devMode, 'AutocompleteSingle', 'Dropdown closed');
  }

  private renderAddIcon() {
    return (
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512">
        <path d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 144L48 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l144 0 0 144c0 17.7 14.3 32 32 32s32-14.3 32-32l0-144 144 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-144 0 0-144z" />
      </svg>
    );
  }

  private renderAddButton() {
    if (!this.addBtn) return null;

    return (
      <button class="input-group-btn add add-btn" role="button" onClick={() => this.itemSelect.emit(this.inputValue)} aria-label="Add item" title="Add item">
        {this.addIcon ? <i class={this.addIcon} /> : this.renderAddIcon()}
      </button>
    );
  }

  private renderClearIcon() {
    return (
      <svg class="fa-icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512">
        <path d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z" />
      </svg>
    );
  }

  private renderClearButton() {
    if (this.removeClearBtn || !this.inputValue) return null;

    return (
      <button class="input-group-btn clear clear-btn" role="button" onClick={this.clearInput} aria-label="Clear input" title="Clear input">
        {this.clearIcon ? <i class={this.clearIcon} /> : this.renderClearIcon()}
      </button>
    );
  }

  private clearInput = () => {
    this.inputValue = '';
    this.filteredOptions = [];
    this.focusedOptionIndex = -1;
    this.selectedOptionIndex = -1;

    this.hasBeenInteractedWith = true;
    this.dropdownOpen = false;

    if (this.required && this.hasBeenInteractedWith) {
      this.validation = true;
    }

    this.clear.emit();
    logInfo(this.devMode, 'AutocompleteSingle', 'Input cleared');
  };

  private handleClickOutside = (event: MouseEvent) => {
    const path = event.composedPath();
    if (!path.includes(this.el)) {
      this.closeDropdown(); // ✅ closes dropdown
    }
  };

  private renderValidationMessages = (ids: string) => {
    if (this.validation && this.validationMessage) {
      if (this.formLayout === 'horizontal' || this.formLayout === 'inline') {
        return (
          <div class="row">
            <div class={{ 'col-sm-2': true, 'col': this.formLayout === 'inline' }}></div>
            <div class={{ 'col-sm-10': true, 'col': this.formLayout === 'inline' }}>
              <div id={`${ids}-validation`} class="invalid-feedback" aria-live="polite">
                {this.validationMessage}
              </div>
            </div>
          </div>
        );
      }
      return <div class="invalid-feedback">{this.validationMessage}</div>;
    }
    return '';
  };

  private renderErrorMessages = (ids: string) => {
    if (this.error && this.errorMessage) {
      if (this.formLayout === 'horizontal' || this.formLayout === 'inline') {
        return (
          <div class="row">
            <div class={{ 'col-sm-2': true, 'col': this.formLayout === 'inline' }}></div>
            <div class={{ 'col-sm-10': true, 'col': this.formLayout === 'inline' }}>
              <div id={`${ids}-error`} class="error-message" aria-live="polite">
                {this.errorMessage}
              </div>
            </div>
          </div>
        );
      }
      return <div class="error-message">{this.errorMessage}</div>;
    }
    return '';
  };

  private renderLayout = (ids: string, names: string) => {
    const isHorizontalOrInline = this.formLayout === 'horizontal' || this.formLayout === 'inline';

    return (
      <div class={{ row: isHorizontalOrInline, [this.formLayout]: isHorizontalOrInline }}>
        {this.renderLabel(ids)}
        <div class={isHorizontalOrInline ? 'col-sm-10' : null}>
          <div
            class={{
              'input-group autocomplete-single-select': true,
              'is-invalid': this.validation,
              [this.validation && this.isFocused ? 'is-invalid-focused' : this.isFocused ? 'ac-focused' : '']: true, // ✅ highlights on input focus
              [`${this.size}`]: !!this.size,
            }}
          >
            {this.renderInputField(ids, names)}
            {this.renderAddButton()}
            {this.renderClearButton()}
          </div>
          {this.renderDropdown(ids)}
        </div>
      </div>
    );
  };

  componentDidLoad() {
    document.addEventListener('click', this.handleClickOutside);
    this.validation = false;
    this.hasBeenInteractedWith = false;
  }

  componentWillLoad() {
    this.validation = false;
    this.hasBeenInteractedWith = false;

    if (!Array.isArray(this.options)) {
      logError(this.devMode, 'AutocompleteSingle', `Expected 'options' to be an array, got ${typeof this.options}`);
    }

    if (!this.label) {
      logWarn(this.devMode, 'AutocompleteSingle', 'Missing label prop; accessibility may be impacted');
    }
  }

  disconnectedCallback() {
    document.removeEventListener('click', this.handleClickOutside);
  }

  public validate(): boolean {
    if (this.required && this.inputValue.trim() === '') {
      this.validation = true;
      return false;
    }
    this.validation = false;
    return true;
  }

  render() {
    const ids = this.camelCase(this.inputId).replace(/ /g, '');
    const names = this.camelCase(this.label).replace(/ /g, '');

    return (
      <div
        class={{
          'autocomplete-container': true,
          'form-group': true,
        }}
      >
        {this.renderLayout(ids, names)}
        {this.renderValidationMessages(ids)}
        {this.renderErrorMessages(ids)}
      </div>
    );
  }
}
