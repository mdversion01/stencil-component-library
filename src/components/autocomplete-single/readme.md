# autocomplete-single



<!-- Auto Generated Below -->


## Properties

| Property            | Attribute            | Description | Type       | Default                      |
| ------------------- | -------------------- | ----------- | ---------- | ---------------------------- |
| `addBtn`            | `add-btn`            |             | `boolean`  | `false`                      |
| `addIcon`           | `add-icon`           |             | `string`   | `''`                         |
| `arialabelledBy`    | `arialabelled-by`    |             | `string`   | `''`                         |
| `clearIcon`         | `clear-icon`         |             | `string`   | `''`                         |
| `devMode`           | `dev-mode`           |             | `boolean`  | `false`                      |
| `disabled`          | `disabled`           |             | `boolean`  | `false`                      |
| `error`             | `error`              |             | `boolean`  | `false`                      |
| `errorMessage`      | `error-message`      |             | `string`   | `''`                         |
| `formId`            | `form-id`            |             | `string`   | `''`                         |
| `formLayout`        | `form-layout`        |             | `string`   | `''`                         |
| `inputId`           | `input-id`           |             | `string`   | `''`                         |
| `label`             | `label`              |             | `string`   | `''`                         |
| `labelHidden`       | `label-hidden`       |             | `boolean`  | `false`                      |
| `options`           | `options`            |             | `string[]` | `[]`                         |
| `placeholder`       | `placeholder`        |             | `string`   | `'Type to search/filter...'` |
| `removeClearBtn`    | `remove-clear-btn`   |             | `boolean`  | `false`                      |
| `required`          | `required`           |             | `boolean`  | `false`                      |
| `size`              | `size`               |             | `string`   | `''`                         |
| `type`              | `type`               |             | `string`   | `''`                         |
| `validation`        | `validation`         |             | `boolean`  | `false`                      |
| `validationMessage` | `validation-message` |             | `string`   | `''`                         |


## Events

| Event            | Description | Type                                                |
| ---------------- | ----------- | --------------------------------------------------- |
| `clear`          |             | `CustomEvent<void>`                                 |
| `componentError` |             | `CustomEvent<{ message: string; stack?: string; }>` |
| `itemSelect`     |             | `CustomEvent<string>`                               |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
