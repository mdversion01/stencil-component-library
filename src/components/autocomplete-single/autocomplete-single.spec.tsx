// src/components/autocomplete-single/autocomplete-single.spec.tsx
import { newSpecPage } from '@stencil/core/testing';
import { AutocompleteSingle } from './autocomplete-single';

describe('<autocomplete-single>', () => {
  it('renders correctly with default props', async () => {
    const page = await newSpecPage({
      components: [AutocompleteSingle],
      html: `<autocomplete-single label="Test Label" input-id="test"></autocomplete-single>`,
    });

    expect(page.root).toBeTruthy();
    expect(page.root.querySelector('input')).toBeTruthy();
    expect(page.root.querySelector('label')?.textContent).toContain('Test Label');

     // âœ… Snapshot rendering
     expect(page.root).toMatchSnapshot();

  });

  it('filters options based on input', async () => {
    const page = await newSpecPage({
      components: [AutocompleteSingle],
      html: `<autocomplete-single label="Fruits" input-id="fruit" placeholder="Type fruit..."></autocomplete-single>`,
    });

    const component = page.rootInstance;
    component.options = ['Apple', 'Banana', 'Orange'];
    component.inputValue = 'Ap';
    component['filterOptions'](); // invoke private method
    await page.waitForChanges();

    expect(component.filteredOptions).toEqual(['Apple']);
  });

  it('emits itemSelect event on selecting option', async () => {
    const page = await newSpecPage({
      components: [AutocompleteSingle],
      html: `<autocomplete-single></autocomplete-single>`,
    });

    const component = page.rootInstance;
    const spy = jest.fn();
    component.options = ['One', 'Two', 'Three'];
    component.itemSelect = { emit: spy } as any;
    component.selectOption('Two');
    expect(spy).toHaveBeenCalledWith('Two');
  });

  it('emits clear event on clearInput', async () => {
    const page = await newSpecPage({
      components: [AutocompleteSingle],
      html: `<autocomplete-single required></autocomplete-single>`,
    });

    const component = page.rootInstance;
    const spy = jest.fn();
    component.clear = { emit: spy } as any;
    component.inputValue = 'Some text';
    component.clearInput();
    expect(component.inputValue).toBe('');
    expect(spy).toHaveBeenCalled();
  });

  it('sanitizes input to prevent HTML injection', async () => {
    const page = await newSpecPage({
      components: [AutocompleteSingle],
      html: `<autocomplete-single></autocomplete-single>`,
    });

    const component = page.rootInstance;
    const malicious = '<script>alert("XSS")</script>';
    const sanitized = component['sanitizeInput'](malicious);
    expect(sanitized).not.toContain('<');
    expect(sanitized).not.toContain('>');
  });
});
