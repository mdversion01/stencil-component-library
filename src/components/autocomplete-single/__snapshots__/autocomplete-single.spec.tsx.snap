// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<autocomplete-single> renders correctly with default props 1`] = `
<autocomplete-single input-id="test" label="Test Label">
  <div class="autocomplete-container form-group">
    <div>
      <label class="form-label" htmlfor="test">
        <span>
          Test Label
        </span>
      </label>
      <div>
        <div class="autocomplete-single-select input-group">
          <input aria-autocomplete="list" aria-controls="test-listbox" aria-expanded="false" aria-haspopup="listbox" aria-labelledby="" aria-required="false" autocomplete="off" class="form-control" id="test" inputmode="text" name="testLabel" placeholder="Type to search/filter..." role="combobox" spellcheck="false" type="" value="">
        </div>
      </div>
    </div>
  </div>
</autocomplete-single>
`;
